### PR TITLE
Apply vscode-python changes from v2024.6.0 to v2024.10.0

### PR DESCRIPTION
**Pull request by wesm**
The latest release has been finalized at https://github.com/microsoft/vscode-python/tree/release/2024.10. 

The new env locator is housed at https://github.com/microsoft/python-environment-tools. The least invasive thing to do for now would be to host builds on GitHub and download then at yarn time like we do for Ark. 

There are also a handful of other changes having to do with the terminal REPL in vscode-python, so we should investigate those to make sure I didn't break anything when resolving the merge conflicts.

### CODE REVIEW

File: extensions/positron-python/src/client/common/process/pythonEnvironment.ts [hunk 1]
```diff
@@ -173,6 +174,23 @@ export async function createCondaEnv(
     return new PythonEnvironment(interpreterPath, deps);
 }
 
+export async function createPixiEnv(
```
Comment (isabelizimm) [inline extensions/positron-python/src/client/common/process/pythonEnvironment.ts:177]: trying things out: we are able to discover pixi environments 🎉 our mechanism for installing `ipykernel` fails, but if you have it already installed, everything seems to work as normal

Screenshot 2024-07-02 at 2 25 15 PM

**Comment by wesm** [comment]
Upstream has finalized their release: https://github.com/microsoft/vscode-python/commits/release/2024.10/. The final commit that isn't in this PR is a no-op (just updating the package.json version), so there is nothing further to apply here other than our changes to deal with the new python-environment-tools Python locator and any other fixes.

**Comment by wesm** [comment]
Rebased.

**Comment by wesm** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by wesm** [comment]
recheck

**Comment by wesm** [comment]
It appears that the new native locator is marked as experimental so it may not be necessary to bundle it since it is disabled by default:

https://github.com/posit-dev/positron/pull/3790/files#diff-ece524e11cd6132207b65f1c98aa0ec1ad098cc54a06369f89410e3526ff6800R581

**Review by petetronic — APPROVED** [review]
Tested Python examples on Mac and major four pane data science functions LGTM.

**Pull request merged**